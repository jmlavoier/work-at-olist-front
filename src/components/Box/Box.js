import React from 'react';
import Logo from 'components/Logo';

import styles from './Box.sass';

const Box = ({ children, ...otherProps }) => (
  <div className={styles.box} {...otherProps}>
    <div className={styles.header}>
      <Logo />
    </div>
    <div>
      {children}
    </div>
  </div>
);

export default Box;
