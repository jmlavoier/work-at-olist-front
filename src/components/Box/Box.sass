@import 'styles/variables'

.box
  border: 3px solid #f2f2f2
  background-color: $color-default-white
  padding: 60px 70px 48px 70px
  font-family: $font-family

.header
  text-align: center
