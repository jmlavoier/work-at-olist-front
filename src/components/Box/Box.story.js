import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Box from './Box';
import Logo from 'components/Logo';

storiesOf('Box', module)
  .add('Default', withInfo()(() => (
    <Box>
      <div style={{ textAlign: 'center' }}>
        <h2>Crie sua conta</h2>
      </div>
    </Box>
  )));
