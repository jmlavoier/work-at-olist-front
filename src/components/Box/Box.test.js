import React from 'react';
import { shallow, mount } from 'enzyme';

import Box from './Box';

describe('<Box />', () => {
  it('Should component render children', () => {
    const child = <div>test</div>;
    const wrapper = shallow(<Box>{child}</Box>);

    expect(wrapper.contains(child)).toBeTruthy();
  });
  it('Should component match snapshot', () => {
    const wrapper = shallow(<Box />);

    expect(wrapper).toMatchSnapshot();
  });
});
