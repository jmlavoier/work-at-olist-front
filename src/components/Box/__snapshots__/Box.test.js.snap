// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> Should component match snapshot 1`] = `
<div
  className="box"
>
  <div
    className="header"
  >
    <Logo />
  </div>
  <div />
</div>
`;
